Add continuous integration

mbe24 opened the pull request:
I've added travis-ci.org for continuous integration. The .travis.yml specifies a build matrix with two builds. The builds per se were sucessfuly, but only 5/12 unit tests passed (cf. https://travis-ci.org/mbe24/boolberry/builds/165003377).

Boost is now used in the highest version found on the system.

I've also added a build status flag to the README. The flag still points to my build. Please create and import this project to travis and use own status flag, if merging this pull request.

Commit:
Create .travis.yml

Commit:
Update README.md

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update CMakeLists.txt

Use highest version of boost

Commit:
Update .travis.yml

1blockologist commented:
You should pin Boost and any dependency to a version because otherwise it makes problems in builds unrepeatable.

Newer doesn't mean better.

Otherwise LGTM, just change boost back to 1.53 or test it with another specific version and I'll merge it.

Commit:
Change boost version

This changes one build to use 1.53 and g++-4.9. 
The second build, using the latest boost version and g++-5, remains untouched.

Commit:
Fix sources

Commit:
Add source for libboost1.53-all-dev

Commit:
Use sourceline for boost1.53 source

Commit:
Update .travis.yml

Commit:
Update .travis.yml

Commit:
Update .travis.yml

1blockologist commented:
LGTM, might revert cmakelists.txt, but I'll monitor the travis-CI builds too

mbe24 commented:
Error seems to be 

```
cc1: error: -Werror=maybe-uninitialized: no option -Wmaybe-uninitialized
cc1: error: unrecognized command line option ‘-std=c11’
```

...What a hassle. At least the build matrix is valid now.
